src/test/resources/features/Branch.feature

```diff
@@ -0,0 +1,22 @@
+Feature: git branch command
+  Background:
+    Given a working directory
+    And a repository .jgit
+
+  Scenario: Create branch without initial state (initial commit)
+    When the user want to execute the command "branch" with argument "new_branch"
+    Then the result is "fatal: Not a valid object in 'HEAD'"
+
+
+  # Does not work because of the bug in the commit command
+  Scenario: Branch already exists
+    Given the user want to execute the command "commit" with argument "My first commit"
+    When the user want to execute the command "branch" with argument "new_branch"
+    And the user want to execute the command "branch" with argument "new_branch"
+    Then the result is "fatal: A branch named 'new_branch' already exists."
+
+  # Does not work because of the bug in the commit command
+  Scenario: Create new branch
+    Given the user want to execute the command "commit" with argument "My first commit"
+    When the user want to execute the command "branch" with argument "dev"
+    Then the result is "Switched to a new branch 'dev'"
```

Comment:
Test à faire

Reply:
Je croyais avoir laissé un commentaire sur la ligne juste au dessus du scénario

Reply:
Ton commentaire indique que le test ne marche pas, mais je n'en vois pas.

Reply:
Tu n'en vois pas comment. Je ne te comprends pas ! Ce scénario marche chez toi quand tu l'exécute ?

Reply:
Non, tu n'as pas de code pour le scenario

Reply:
Exactement. Je passe par le main donc j'écris pas vraiment de code pour le scénario. Quand le commit sera bon en passant par main, je pourrai compléter certaines choses. Par exemple `@Given('a file stored on disk with content {string}')`, `@When the user add {string} in the file ...`, plus de variable file ou folder local à chaque StepDefs. Pour moi, avec ce genre de test case, on aura plus le probleme de wd. Mbon, tout ça n'est que théorique.

Reply:
En voyant le commentaire sur tes scénarios je pense que tu as peur de créer des bugs en insérant des tests qui ne peuvent pas encore marcher car Commit n'est pas fini (il semblerait?). C'est tout le but de préparer ses tests en avance ! En fait si tu t'accordes avec Fayssal, tu peux t'assurer de quel sera le résultat final et ainsi coder tes tests comme s'ils fonctionnaient. Cela devrait t'éviter d'avoir du travail en plus plus tard, car si les tests préparés ne marchent pas c'est qu'une nouvelle PR a tout cassé donc ce n'est pas ton job de chercher la source du problème ✨

Reply:
ok